TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -1445,9 +1446,53 @@ currentIndex < (lastIndex+1) && (line[currentIndex + 2] == '"' || line[currentIn
-                    if(nextCharIsACobolWordChar && line[currentIndex] == '-')
+                    if (currentState.LastSignificantToken != null && (currentState.LastSignificantToken.TokenType == TokenType.PERFORM || currentState.LastSignificantToken.TokenType == TokenType.THRU || currentState.LastSignificantToken.TokenType == TokenType.THROUGH || currentState.LastSignificantToken.TokenType == TokenType.OF || currentState.LastSignificantToken.TokenType == TokenType.IN)
+                        && nextCharIsDashFollowedByCobolChar)
+                    {
+                        // Reset scanner state
+                        currentIndex = saveCurrentIndex;
+                        if (numericLiteralToken.TokenType == TokenType.InvalidToken)
+                        {
+                            tokensLine.ClearDiagnosticsForToken(numericLiteralToken);
+                        }
+                        //UserDefinedWord or PartialCobolWord
+                        return ScanCharacterString(startIndex);
+                    }
+
+                    //I use tokensLine.ScanState.AtBeginningOfSentence to match what is done by method ScanKeywordOrUserDefinedWord called by ScanCharacterString
+                    //
+                    //TODO check if the token is in area A
+                    //But beware, method ScanIsolatedToken in the Scanner use a Token with ColumnsLayout.FreeTextFormat. See TokensLine.CreateVirtualLineForInsertedToken(0, tokenText);
+                    //TODO create an issue for that
+                    //
+                    //Also ScanIsolatedToken receive a string which contains the text without the original spaces before the text. It means the new token generated is align to the left. The original column is lost.
+                    //E.g. if original line is move a to :b:, with a replace like ==:b:== by ==b== method ScanIsolatedToken receive string "b".
+                    //I will create an issue to discuss about it, I think it would be better that the Scanner receive a String with correct preceding whitespaces before the text to prase so column is correct.
+                    else if (tokensLine.ScanState.InsideProcedureDivision && tokensLine.ScanState.AtBeginningOfSentence && nextCharIsDashFollowedByCobolChar)
+                    {
+                        // Reset scanner state
+                        currentIndex = saveCurrentIndex;
+                        if (numericLiteralToken.TokenType == TokenType.InvalidToken)
+                        {
+                            tokensLine.ClearDiagnosticsForToken(numericLiteralToken);
+                        }
+                        //UserDefinedWord or PartialCobolWord
+                        return ScanCharacterString(startIndex);
+                    }
+                    //TODO this case should dissappear. This case only handles numericLiteralToken followed by a dash and (a letter or an underscore). E.g. "100-X"
```

Comment:
'disappear'

Reply:
With this PR the following code will fail:

```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. Pgm.
      
       PROCEDURE DIVISION.
           perform 300.
         300.
           exit.
       END PROGRAM Pgm.      
```

Reply:
Yes it fails, but `perform 300` is out of the scope of #2167 and even #1982.
I'll create an issue for this case and I also need to check if our code base use this syntax.